### ISSUE TITLE
One hot encoding

### ISSUE THREAD

**Issue by mavaylon1** [divamgupta/image-segmentation-keras#261]
From the blog post it seems the ground truths are not one-hot encoded. Can you explain why not?

**Comment by divamgupta** [comment]
They are converted to one hot by the dataloader

**Comment by mavaylon1** [comment]
I couldn’t find where that is. Looking at get_segmentation_array it returns this “ seg_labels = np.reshape(seg_labels, (width*height, nClasses))”.
That shape isn’t one hot encoded

**Comment by mavaylon1** [comment]
You have no_reshape=False so it will always be reshaped since you used an "if not" statement. If we set that to true then I can see it being one-hot, but it's never set to true in train.py or data_loader.py

**Comment by darshats** [comment]
Hello,
I have the same question. Since default value of param is false, it will always end up reshaping. For multi category segmentation, do you recommend users set it to true?

Also I am curious that images are converted to channel first. Maybe I missed it, but are masks converted to channel first?

**Comment by darshats** [comment]
@divamgupta  - Also could you help understand in which scenarios the mask needs to be flattened, shape=(width*height, channels)? Kind of goes against the readme that says for every pixel in the image, the corresponding pixel in the mask is the label.

**Comment by divamgupta** [comment]
While preprocessing you need to make the masks as described in the readme. The data generator will automatically reshape then to (width*height, channels) to fit the model output shape.

**Comment by darshats** [comment]
Thank you Divam for taking the time to reply. I also appreciate that this repository is constantly kept updated, kudos on that!

wrt the above, in case of segmentation like unet isnt the output the same shape as input. Then I am still unsure when the no_reshape flag should be set to True.

**Comment by divamgupta** [comment]
Yes the input and output are of the same shape before the reshape step. But
users don’t really need to worry about this reshape as both the model and
data generators reshape the output. The output of the model is reshaped in
the get_segmentation_model function.

**Comment by mavaylon1** [comment]
Why do we have the reshaping to (width*height, channels) in the first place? I understand the it is reshaped to that both before and after the model. But why?
